src/mscorlib/corefx/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -2808,6 +2801,23 @@ void InsertHash(TokenHashValue[] hashTable, String str, TokenType tokenType, int
             } while (i < TOKEN_HASH_SIZE);
             Contract.Assert(true, "The hashtable is full.  This should not happen.");
         }
+
+        private bool CompareStringIgnoreCaseOptimized(string string1, int offset1, int length1, string string2, int offset2, int length2)
+        {
+            if (length1 != length2)
```

Comment:
re-adding the comment: this is not correct assumptions. you can linguistically have 2 string with different lengths and still be equal. 

I think if we need to optimize here for the length==1 cases we can do it like 

if (length1 == 1 && length2 == 1 && string1[offset1] == string2[offset2])
   // return true
else 
// do the slow path by CompareInfo call.

Note that I didn't compare the characters in the condition as case insensitive because the main stream cases we'll have characters not really need casing which is the scenario we really need to optimize for